fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/JoinAssociativityRule.java

```diff
@@ -57,6 +62,20 @@ public boolean check(final OptExpression input, OptimizerContext context) {
             return Collections.emptyList();
         }
 
+        if (leftChildJoin.getProjection() != null) {
+            Projection projection = leftChildJoin.getProjection();
+
+            for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : projection.getColumnRefMap().entrySet()) {
+                if (!entry.getValue().isColumnRef()) {
+                    return Collections.emptyList();
+                }
+
+                if (!entry.getKey().equals(entry.getValue())) {
+                    return Collections.emptyList();
+                }
+            }
+        }
```

Comment:
Why forbidden associativite Project-Join?

Reply:
When this PR has an expression directly in the JOIN, the original logic is used, and no join reorder is performed